scripts/o3de/ExportScripts/export_source_ios_xcode.py

```diff
@@ -0,0 +1,122 @@
+
+        process_command(f"cmake --build {tools_build_folder_str} --target Editor AssetProcessorBatch --config profile".split(),
+                    cwd=ctx.project_path)
+        
+    # Optionally process the assets
+    if should_process_assets:
+        asset_processor_batch_path = tools_build_folder / 'bin/profile/AssetProcessorBatch'
+        process_command([ str(asset_processor_batch_path), '--platforms=ios' ,
+                        '--project-path', ctx.project_path ], cwd=ctx.project_path)
+
+    #generate the xcode project file for the O3DE project
+    cmake_toolchain_path = ctx.engine_path / 'cmake/Platform/iOS/Toolchain_ios.cmake'
+
+    process_command(['cmake', '-B', ios_build_folder_str, '-G', "Xcode", f'-DCMAKE_TOOLCHAIN_FILE={str(cmake_toolchain_path)}', '-DLY_UNITY_BUILD=ON', '-DLY_MONOLITHIC_GAME=1'],
+                    cwd= ctx.project_path)
+
+    if logger:
+        logger.info(f"XCode project file should be generated now. Please check {ios_build_folder_str}")
+
+
+
+# This code is only run by the 'export-project' O3DE CLI command
+if "o3de_context" in globals():
+    global o3de_context
+    global o3de_logger
+
+    def parse_args(o3de_context: exp.O3DEScriptExportContext):
+        parser = argparse.ArgumentParser(
+                    prog=f'o3de.py export-project -es {__file__}',
+                    description="Exports a project as a generated iOS Xcode project in the project directory. "
+                                "In order to use this script, the engine and project must be setup and registered beforehand. ",
+                    epilog=exp.CUSTOM_CMAKE_ARG_HELP_EPILOGUE,
+                    formatter_class=argparse.RawTextHelpFormatter,
+                    add_help=False
+        )
+
+        default_tools_path = o3de_context.project_path / 'build/Mac'
+        default_ios_path = o3de_context.project_path / 'build/iOS'
+        parser.add_argument(exp.CUSTOM_SCRIPT_HELP_ARGUMENT,default=False,action='store_true',help='Show this help message and exit.')
+        parser.add_argument('-bt', '--build-tools', default=True, action='store_true',
+                            help="Specifies whether to build O3DE toolchain executables. This will build AssetBundlerBatch, AssetProcessorBatch.")
+        parser.add_argument('-tbp', '--tools-build-path', type=pathlib.Path, default=default_tools_path,
+                                help='Designates where the build files for the O3DE toolchain are generated. If not specified, default is <o3de_project_path>/build/tools.')
+        parser.add_argument('-ibp', '--ios-build-path', type=pathlib.Path, default=default_tools_path,
+                                help='Designates where the build files for the O3DE toolchain are generated. If not specified, default is <o3de_project_path>/build/tools.')
+        parser.add_argument('-assets', '--should-process-assets', default=True, action='store_true',
```

Comment:
Having `store_true` and `default=True` makes this argument useless. There is no way to set to false.